### PR TITLE
Rewrite activity in pygame for better animations and UI

### PR DESCRIPTION
**Pull request by sparshg**
@chimosky @srevinsaju
~~Its still work in progress~~

### CODE REVIEW

**Comment by srevinsaju** [comment]
Can you remind me if we had discussed on this earlier / issue number? What are the issues we plan to solve with pygame?

**Comment by sparshg** [comment]
In pygame we could add nice animations and also make overall UI better, we had discussed about this on a meet previously.

File: TestActivity.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#
```
Comment (chimosky) [inline TestActivity.py]: Please rename this file.

File: anim.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import pygame as pg
```
Comment (chimosky) [inline anim.py:17]: Could you add a license to this and other files you created? Thanks!

File: main.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+from asyncio import constants
+import pygame as pg
+from constants import *
+from frame import Frame
+import sys
+
+# The main controller
```
Comment (chimosky) [inline main.py:28]: Leave an extra line before any comment or class definition.

**Comment by chimosky** [comment]
@sparshg State the advantage in your commit message and any addition that prompted the rewrite.

**Comment by sparshg** [comment]
@chimosky Please review till here, I tried adding a license to my files, fixed previous commit messages and tested on sugar live build. Still have to add a few features.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,160 @@
+# Byte-compiled / optimized / DLL files
```
Comment (chimosky) [inline .gitignore]: This is a very exhaustive .gitignore, why is it this way?

**Comment by chimosky** [comment]
Tested 80fe103, works as expected.

What features are you working on?

It'll be nice to have a welcome page that explains a bit about the game like how to play it and any other info that's necessary.

This seems like a great activity to have collaboration so two players on different laptops can play, you can take a look at [collabwrapper](https://github.com/samdroid-apps/collabwrapper) and this [search](https://github.com/search?l=Python&q=org%3Asugarlabs+collabwrapper&type=Code) to see examples of implementations, feel free to ask any questions if you decide to work on this.

**Comment by sparshg** [comment]
Currently I am working on the UI elements, title score and a reset button. I don't think that info page is really necessary as this game is known to everyone, but I will add a help menu still that displays instructions.
Regarding the collabwrapper, I will try to open another PR for that once I complete the work for some other activities.

**Comment by sparshg** [comment]
@chimosky Please review, I have completed this PR and tested extensively, everything seems to be stable and smooth now. Also working fine on live build.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+# Cython debug symbols
+cython_debug/
+
+# PyCharm
+#  JetBrains specific template is maintained in a separate JetBrains.gitignore that can
+#  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
+#  and can be added to the global gitignore or merged into this file.  For a more nuclear
+#  option (not recommended) you can uncomment the following to ignore the entire idea folder.
+#.idea/
```
Comment (chimosky) [inline .gitignore]: Could you fix this?

File: main.py [hunk 6]
```diff
@@ -32,31 +31,30 @@ def check_events(self):
                 pg.display.set_mode(event.size, pg.RESIZABLE)
                 break
             if event.type == pg.MOUSEBUTTONUP:
-                if self.canvas is None:
-                    self.canvas.grab_focus()
+                # if self.canvas is None:
+                #     self.canvas.grab_focus()
```
Comment (chimosky) [inline main.py]: You can delete these lines instead of commenting them out.

File: frame.py [hunk 7]
```diff
@@ -107,6 +107,10 @@ def check(sum, i, j):
                     self.moves[i][j].blink = True
                     self.moves[i][j].blink_start = pg.time.get_ticks()
                     i, j = prev
+                if sum == 3:
```
Comment (chimosky) [inline frame.py:113]: Why 3?

File: .gitignore (hunk 4 above)
Comment (sparshg) [reply .gitignore]: I copied GitHub's python gitignore template, will fix that

**Comment by chimosky** [comment]
Noticed you added the help in a `DescriptionItem`, I think the better way would be to use a button with a help icon where the text isn't editable, you can see how [implode activity](https://github.com/sugarlabs/implode-activity/blob/master/implodeactivity.py#L230) added the help button and you can just make it display same text but not editable.

**Comment by chimosky** [comment]
Tested c63d0f3.

**Comment by sparshg** [comment]
@chimosky I fixed the issues and added custom help button on the top right. Also made a demo image for the page. Please test.

**Comment by chimosky** [comment]
Tested e255c19, works as expected.

I'll wait for @srevinsaju to test and review before merging.

File: frame.py (hunk 7 above)
Comment (sparshg) [reply frame.py:113]: If there are three crosses in a row/col/diagonal, so assuming cross has a value of 1, sum =3, and circles have value -1, so sum = -3

Comment (chimosky) [reply frame.py:113]: Thanks!

**Pull request merged**